### ISSUE TITLE
Should dynamically calculated property values (like `scope` and `start_url`) be programmatically accessible?

### ISSUE THREAD

**Issue by tomayac** [w3c/manifest#931]
Something I see as lot of developers trip over is `scope` handling (`start_url` not so much, since it is easier to detect when it does not work as intended). However, developers repeatedly fall for unexpected `scope`s, for example, when they host their manifest from a sub folder.

Would it make sense to expose the [calculated `scope`](https://w3c.github.io/manifest/#ref-for-dfn-application-context-12:~:text=The%20steps%20for%20processing%20the%20scope%20member) via JavaScript (something like `document.manifest.scope`)? This would allow developers to debug and check the `scope`, to, for example, programmatically handle out-of-scope links differently (maybe to add an icon or change the screen orientation,…).

**Comment by dominickng** [comment]
I think it makes more sense for browser DevTools to display the calculated scope (e.g. Chrome could do so in Application > Manifest) rather than expose the computed value in JS.

Since the developer can explicitly set the scope, ideally they wouldn't have to query it dynamically.

**Comment by alancutter** [comment]
Another use case for read access to the browser's view of the manifest is feature detection. E.g. a site can tell whether the browser has file handler support by whether their file_handlers field is present.

**Comment by tomayac** [comment]
> I think it makes more sense for browser DevTools to display the calculated scope (e.g. Chrome could do so in Application > Manifest) rather than expose the computed value in JS.

Absolutely. Filed https://crbug.com/1108717. 

> Since the developer can explicitly set the scope, ideally they wouldn't have to query it dynamically.

The mentioned use cases from the original post and https://github.com/w3c/manifest/issues/931#issuecomment-662814210 would still be facilitated by providing the data programmatically rather than requiring developers to calculate them. Similar to a service worker's scope, which is exposed [1] despite being calculable. 

——
`(await navigator.serviceWorker.ready).scope`

**Comment by marcoscaceres** [comment]
I agree, this seems more like a DevTools thing - where the computed scope should be shown. Dev tools could provide a way to check if another URL is in scope.

**Comment by tomayac** [comment]
FWIW, @rayankans is exploring exposing Web App Manifest data on [`navigator.shortcuts.getAll()`](https://github.com/rayankans/app-shortcuts/blob/master/IDL.md#readme:~:text=getAll()%3B).

**Comment by marcoscaceres** [comment]
going to defer on this, at it's a feature request. Having said that, it might be ok to expose the shortcut data, but anything further should really happen in this spec in the future.

**Comment by mgiuca** [comment]
It's interesting, @alancutter suggested this exact feature a few weeks ago as a solution to our internal testing of the Manifest spec.

We wouldn't necessarily want to add an API just for testing, but if we had other uses cases like the above, it might serve that role.
